creator-node/src/FileProcessingQueue.js

```diff
@@ -0,0 +1,93 @@
+const Bull = require('bull')
+const { logger: genericLogger } = require('./logging')
+const config = require('./config')
+const redisClient = require('./redis')
+const { handleTrackContentRoute: transcodeFn } = require('./components/tracks/tracksComponentService')
+const { serviceRegistry } = require('./serviceRegistry')
+
+const EXPIRATION = 7200 // 2 hours in seconds
+const PROCESS_NAMES = Object.freeze({
+  transcode: 'transcode'
+})
+const PROCESS_STATES = Object.freeze({
+  IN_PROGRESS: 'IN_PROGRESS',
+  DONE: 'DONE',
+  FAILED: 'FAILED'
+})
+
+class FileProcessingQueue {
+  constructor () {
+    this.queue = new Bull(
+      'fileProcessing', {
+        redis: {
+          host: config.get('redisHost'),
+          port: config.get('redisPort')
+        },
+        defaultJobOptions: {
+          removeOnComplete: true,
+          removeOnFail: true
+        }
+      }
+    )
+    this.maxConcurrency = 5 // arbitrary; can adjust
+
+    this.queue.process(PROCESS_NAMES.transcode, this.maxConcurrency, async (job, done) => {
+      const { transcodeParams } = job.data
+      const resp = await this._trackProgress(PROCESS_NAMES.transcode, transcodeFn, transcodeParams)
+      done(resp)
```

Comment:
we should try catch here and call done() with the error if this throws (see some of the other queues)